src/expressions.js

```diff
@@ -67,16 +67,22 @@ exports.removeGivenScopes = function(scopeset, expr) {
   }
 
   if (expr.AllOf) {
-    const AllOf = expr.AllOf.map(e => exports.removeGivenScopes(scopeset, e)).filter(e => e !== null);
+    const AllOf = expr.AllOf.map(e => exports.removeGivenScopes(scopeset, e, false)).filter(e => e !== null);
     if (AllOf.length === 0) {
       return null;
     }
+    if (AllOf.length === 1 && (!topLevel || typeof AllOf[0] !== 'string')) {
+      return AllOf[0];
+    }
     return {AllOf};
   }
 
-  const AnyOf = expr.AnyOf.map(e => exports.removeGivenScopes(scopeset, e));
+  const AnyOf = expr.AnyOf.map(e => exports.removeGivenScopes(scopeset, e, false));
   if (AnyOf.some(e => e === null)) {
     return null;
   }
+  if (AnyOf.length === 1 && (!topLevel || typeof AnyOf[0] !== 'string')) {
```

Comment:
Given `{AnyOf: [{AllOf: ['a', 'b']}]}`, it seems like this `topLevel`` check would refuse to reduce to `{AllOf: ['a', 'b']}`.  Is that the intent?

Reply:
I think that's wrong...

It's only: `{AnyOf: ['a']}` that won't be reduced to `'a'`, isn't that correct? (I think that's okay, not sure... I'm not sure I can remember why we wanted that, but it seems okay to me regardless)

Reply:
You're right.  But why exclude that case?

Reply:
I did this check to try to keep the returned thing an expression. I felt like returning just `'a'` would break the contract we've set out for this function and make it harder to use. Maybe it is worth doing though! I'm happy either way.